bap-vibes/src/core_c.ml

```diff
@@ -143,262 +139,240 @@ module Eval(CT : Theory.Core) = struct
       hvars;
     }
 
-  let char_ty = Theory.Bitv.define 8
-  let short_ty = Theory.Bitv.define 16
-  let long_ty = Theory.Bitv.define 32
-  let long_long_ty = Theory.Bitv.define 64
+  let char_ty = T.Bitv.define 8
+  let short_ty = T.Bitv.define 16
+  let long_ty = T.Bitv.define 32
+  let long_long_ty = T.Bitv.define 64
 
-  let ty_of_base_type info (c_ty : base_type) : _ T.Bitv.t T.Value.sort =
+  let ty_of_base_type
+      (info : _ interp_info)
+      (c_ty : Patch_c.typ) : _ T.Bitv.t T.Value.sort =
     match c_ty with
-    | CHAR _ -> char_ty
-    | INT (SHORT, _) -> short_ty
-    | INT (LONG, _) -> long_ty
-    | INT (LONG_LONG, _) -> long_long_ty
-    | _ -> info.word_sort
-
-  let ty_op_pointer_type info (c_ty : base_type) : _ T.Bitv.t T.Value.sort =
+    | VOID -> T.Bitv.define 0
+    | INT (`r8, _) -> char_ty
+    | INT (`r16, _) -> short_ty
+    | INT (`r32, _) -> long_ty
+    | INT (`r64, _) -> long_long_ty
```

Comment:
Maybe it would be nice to translate FrontC types to this datatype. The PatchC datatypes are meant to be greatly simplified (to the point where it's trivial to translate to Core Theory sorts). Perhaps this can be its own PR.